src/backend/commands/explain.c

```diff
@@ -1203,72 +1203,36 @@ ExplainNode(PlanState *planstate, List *ancestors,
 	char       *skip_outer_msg = NULL;
 	int			motion_recv;
 	int			motion_snd;
-	float		scaleFactor = 1.0; /* we will divide planner estimates by this factor to produce
-									  per-segment estimates */
-	Slice		*parentSlice = NULL;
+
+	/*
+	 * We will divide planner estimates by this factor to produce per-segment
+	 * estimates. In those cases when a command is executed on a single
+	 * segment, save_currentSlice->gangSize will equal 0, so we use
+	 * scaleFactor == 1 instead. save_currentSlice can be null if we use
+	 * utility mode.
+	 */
+	float		scaleFactor = (save_currentSlice && save_currentSlice->gangSize) ?
+	(float) save_currentSlice->gangSize :
+	1.0;
 
 	/* Remember who called us. */
 	parentplanstate = es->parentPlanState;
 	es->parentPlanState = planstate;
 
-	if (Gp_role == GP_ROLE_DISPATCH)
+	/* Motion nodes are handled separately */
+	if (Gp_role == GP_ROLE_DISPATCH && !IsA(plan, Motion))
```

Comment:
Can you explain, why do you make exception for Motions here?

Reply:
I removed this block of code because it didn't affect on scaleFactor